### PR TITLE
Simplest converter for GSI obs error correlation files

### LINKED ISSUE #586: ioda-converter for GSI files specifying cross-channel correlations
## Description

Write a converter from GSI files specifying cross-channel correlations to JEDI files specifying cross-channel correlations.

### PR DESCRIPTION
**Pull request by shlyaeva**
## Description

This is a first version of the converter for GSI obs error cross-channel correlation files.
It's hacky in two ways:
- uses hacked GSI files (they need to store channel numbers instead of channel indices)
- it doesn't use ioda-engines interface to write out files. The input files are Fortran-written binary sequential files, and it's a pain to read them in C++ (although I have some tips from Ryan, so I can work on this more and improve the converter. I just wanted to have something ready sooner rather than later).

### Issue(s) addressed

- fixes #586

### CODE REVIEW

File: src/obserror/obserror2ioda.f90 [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+integer :: ncid_out, nch_dimid, varid_nch, varid_corr
+integer, allocatable, dimension(:)   :: indxR  !< (nch_active)
+real(4), allocatable, dimension(:,:) :: readR4 !< (nch_active, nch_active)
+
+!> get filenames from command line arguments
+call getarg(1,filename_in)
+call getarg(2,filename_out)
+
+!> read GSI obs error covariances
+open(lu,file=trim(filename_in),convert='little_endian',form='unformatted')
```
[the remark this answers is not in the document]
Comment (shlyaeva) [reply src/obserror/obserror2ioda.f90:22]: I think it should be alright. I've copy-pasted this code from GSI, which I think means they always use `little_endian` for those files. @KristenBathmann-NOAA is that correct?

**Comment by shlyaeva** [comment]
> These errors should have been fixed with #609. I notice that this branch is out of sync with develop, so perhaps that's the issue. I'll hit the merge develop button now and hopefully these errors will go away.

Thank you for fixing this!

File: src/obserror/obserror2ioda.f90 (hunk 1 above)
[the remark this answers is not in the document]
Comment (KristenBathmann-NOAA) [reply src/obserror/obserror2ioda.f90:22]: Yes these files are little endian

**Comment by KristenBathmann-NOAA** [comment]
Does the accuracy of the covariances in the binary file matter at this time? I realized that when I supplied the file with different channel indices that I over-inflated some of the diagonal values. I can provide a new one if needed.

**Comment by shlyaeva** [comment]
> Does the accuracy of the covariances in the binary file matter at this time? I realized that when I supplied the file with different channel indices that I over-inflated some of the diagonal values. I can provide a new one if needed.

I think it doesn't matter in this repository, since the file here is only used to test the converter, and the values don't matter in this case.

It probably also isn't crucial for the ufo test, since it's only testing I/O and inverse. It would be good to have the correct one for the tests with Var.

File: src/obserror/obserror2ioda.f90 (hunk 1 above)
[the remark this answers is not in the document]
Comment (srherbener) [reply src/obserror/obserror2ioda.f90:22]: Thanks for the feedback about the file "ended-ness". I think this is good for now. We can always address the ended-ness later if the situation ever changes.

**Review by srherbener — APPROVED** [review]

**Review by KristenBathmann-NOAA — APPROVED** [review]
Looks good.

**Pull request merged**